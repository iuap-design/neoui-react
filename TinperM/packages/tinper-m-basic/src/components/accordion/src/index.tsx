import './Accordion.less'
import { attachPropertiesToComponent } from '@utils/AttachPropertiesToComponent'
import Accordion, { AccordionPanel } from './Accordion'

export type { AccordionProps, AccordionPanelProps } from './iAccordion'

export default attachPropertiesToComponent(Accordion, { Panel: AccordionPanel })
