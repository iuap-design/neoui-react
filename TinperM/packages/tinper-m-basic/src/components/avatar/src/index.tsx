import './Avatar.less'
import { attachPropertiesToComponent } from '@utils/AttachPropertiesToComponent'
import Avatar from './Avatar'
import AvatarGroup from './AvatarGroup'

export type { AvatarProps } from './iAvatar'

export default attachPropertiesToComponent(Avatar, { Group: AvatarGroup, })
