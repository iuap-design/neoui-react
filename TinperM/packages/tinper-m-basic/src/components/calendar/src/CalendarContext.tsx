import { createContext, PropsWithChildren, useContext } from 'react';

import { DateRange, Page } from './Convert';
import { MergedCalendarProps } from '@components/calendar/src/Calendar';

export type DateType = 'month' | 'year';
export type DateAction = 'subtract' | 'add';



export type CalendarRef = {
  jumpTo: (page: Page | ((page: Page) => Page)) => void
  jumpToToday: () => void
}


type ContextProps = MergedCalendarProps & {
  classPrefix: string
  current: Date
  dateRange: DateRange
  setCurrent: (d: Date) => void
  setDateRange: (d: DateRange) => void

}



const calendarContext = createContext<ContextProps | null>(null)

export const CalendarProvider = (props: PropsWithChildren<ContextProps>) => (
  <calendarContext.Provider value={props}>
    {props.children}
  </calendarContext.Provider>
)

export const useCalendarContext = () => {
  const object = useContext(calendarContext)
  if (!object) { throw new Error('useCalendarContext must be used within a Provider') }
  return object;
}


