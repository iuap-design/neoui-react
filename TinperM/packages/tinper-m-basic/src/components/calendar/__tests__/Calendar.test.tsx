/**Calendar.tsx */
import {render, renderHook, screen} from "@testing-library/react";
import {Calendar} from "@tinper/m";
import React, {useCallback, useId} from "react";
import userEvent from "@testing-library/user-event";
import {getDate, isToday, startOfMonth, startOfWeek} from "date-fns";
import {CalendarRef} from "../src/Calendar";

describe('Calendar Component', () => {


  describe('<test prop:: prevMonthButton>, <test prop:: nextMonthButton>, <test prop:: prevYearButton>, <test prop:: nextYearButton>', () => {
    it('should config prevMonth, nextMonth, prevYear and nextYear for header button', () => {
      const prevMonthText = '上一月';
      const nextMonthText = '下一月';
      const prevYearText = '上一年';
      const nextYearText = '下一年';

      render(<Calendar
        prevMonthButton={<span>{prevMonthText}</span>}
        nextMonthButton={<span>{nextMonthText}</span>}
        prevYearButton={<span>{prevYearText}</span>}
        nextYearButton={<span>{nextYearText}</span>}
      />)

      expect(screen.getByText(prevMonthText)).toBeInTheDocument();
      expect(screen.getByText(nextMonthText)).toBeInTheDocument();
      expect(screen.getByText(prevYearText)).toBeInTheDocument();
      expect(screen.getByText(nextYearText)).toBeInTheDocument();
    });

  });

  describe('<test prop:: selectionMode>', () => {
    it('should select one day when mode is single', async () => {
      const user = userEvent.setup()

      render(<Calendar
        selectionMode='single'
      />)
      await user.click(screen.getByRole('gridcell', {name: /18/i}))

      const selectedCell = screen.getByRole('gridcell', {selected: true})
      expect(selectedCell).toBeInTheDocument();
      expect(selectedCell).toHaveTextContent('18')
    });
    it('should select multiple days when mode is range', async () => {
      const user = userEvent.setup()

      render(<Calendar
        selectionMode='range'
      />)
      await user.click(screen.getByRole('gridcell', {name: /14/i}))
      await user.click(screen.getByRole('gridcell', {name: /15/i}))

      const selectedCells = screen.getAllByRole('gridcell', {selected: true})
      expect(selectedCells).toHaveLength(2)
      expect(selectedCells[0]).toHaveTextContent('14')
      expect(selectedCells[1]).toHaveTextContent('15')
    });

  });

  describe('<test prop:: weekStartsOn>', () => {
    it('should handle render Monday as the first day of week', () => {
      render(<Calendar weekStartsOn="Monday"/>)

      const result = screen.getAllByRole('gridcell')
      const expectedResult = getDate(startOfWeek(startOfMonth(new Date()), {weekStartsOn: 1}))
      expect(result[0]).toHaveTextContent(expectedResult.toString())
    });

  });

  describe('<test prop:: renderLabel>', () => {
    it('should handle custom label rendering', () => {
      render(<Calendar renderLabel={(date: Date) => {
        if (isToday(date)) return '今天'
      }}/>)

      expect(screen.getByRole('gridcell', {
        name: `${new Date().getDate()} 今天`
      })).toBeInTheDocument()
    });

  });
  describe('<test prop:: min>, <test prop:: max>', () => {
    it('should disabled outside of range from min to max', () => {
      const min = new Date()
      const max = new Date()
      min.setDate(12)
      max.setDate(20)

      render(<Calendar min={min} max={max}/>)

      expect(screen.getByRole('gridcell', {name: /21/i})).toHaveClass('mui-calendar-cell-disabled')
      expect(screen.getByRole('gridcell', {name: /11/i})).toHaveClass('mui-calendar-cell-disabled')
      expect(screen.getByRole('gridcell', {name: /12/i})).not.toHaveClass('mui-calendar-cell-disabled')
      expect(screen.getByRole('gridcell', {name: /20/i})).not.toHaveClass('mui-calendar-cell-disabled')
    });

  });

  describe('<test prop:: minPage>, <test prop:: maxPage>', () => {
    const today = new Date()
    const currentYear = today.getFullYear()
    const currentMonth = today.getMonth() + 1

    it('should disabled next month button when maxPage month equals to current, <test prop:: minPage>, <test prop:: maxPage>', async () => {
      const user = userEvent.setup()
      const minPage = {year: currentYear - 1, month: currentMonth}
      const maxPage = {year: currentYear, month: currentMonth + 1}

      render(<Calendar minPage={minPage} maxPage={maxPage}/>)

      const nextMonthButton = screen.getByRole('button', {name: /go to next month/i})

      expect(nextMonthButton).not.toHaveClass('mui-calendar-arrow-button-disabled')

      await user.click(nextMonthButton)

      expect(nextMonthButton).toHaveClass('mui-calendar-arrow-button-disabled')
      const expectedResult = `${currentYear}年${currentMonth + 1}月`
      expect(screen.getByText(expectedResult)).toBeInTheDocument()
    });

    it('should disabled next year button when maxPage year equals to current', async () => {
      const user = userEvent.setup()
      const minPage = {year: currentYear - 1, month: currentMonth}
      const maxPage = {year: currentYear, month: currentMonth + 1}

      render(<Calendar minPage={minPage} maxPage={maxPage}/>)

      const nextYearButton = screen.getByRole('button', {name: /go to next year/i})

      expect(nextYearButton).toHaveClass('mui-calendar-arrow-button-disabled')

      await user.click(nextYearButton)

      expect(nextYearButton).toHaveClass('mui-calendar-arrow-button-disabled')
      const expectedResult = `${currentYear}年${currentMonth}月`
      expect(screen.getByText(expectedResult)).toBeInTheDocument()
    });

    it('should disabled prev month button when minPage month equals to current', async () => {
      const user = userEvent.setup()
      const minPage = {year: currentYear, month: currentMonth - 1}
      const maxPage = {year: currentYear, month: currentMonth + 1}

      render(<Calendar minPage={minPage} maxPage={maxPage}/>)

      const prevMonthButton = screen.getByRole('button', {name: /go to prev month/i})

      expect(prevMonthButton).not.toHaveClass('mui-calendar-arrow-button-disabled')

      await user.click(prevMonthButton)

      expect(prevMonthButton).toHaveClass('mui-calendar-arrow-button-disabled')
      const expectedResult = `${currentYear}年${currentMonth - 1}月`
      expect(screen.getByText(expectedResult)).toBeInTheDocument()
    });

    it('should disabled prev year button when minPage year equals to current', async () => {
      const user = userEvent.setup()
      const minPage = {year: currentYear - 1, month: currentMonth}
      const maxPage = {year: currentYear, month: currentMonth + 1}

      render(<Calendar minPage={minPage} maxPage={maxPage}/>)

      const prevYearButton = screen.getByRole('button', {name: /go to prev year/i})

      expect(prevYearButton).not.toHaveClass('mui-calendar-arrow-button-disabled')

      await user.click(prevYearButton)

      expect(prevYearButton).toHaveClass('mui-calendar-arrow-button-disabled')
      const expectedResult = `${currentYear - 1}年${currentMonth}月`
      expect(screen.getByText(expectedResult)).toBeInTheDocument()
    });
  });

  describe('<test prop:: onPageChange>', () => {
    it.each([
      {buttonName: 'next month', expectedYear: 2024, expectedMonth: 4, description: 'current year and next month'},
      {buttonName: 'next year', expectedYear: 2025, expectedMonth: 3, description: 'next year and current month'},
      {buttonName: 'prev month', expectedYear: 2024, expectedMonth: 2, description: 'current year and prev month'},
      {buttonName: 'prev year', expectedYear: 2023, expectedMonth: 3, description: 'prev year and current month'},
    ])('should called onPageChange with $description', async ({ buttonName, expectedYear, expectedMonth }) => {
      const onPageChange = jest.fn()
      const user = userEvent.setup()
      render(<Calendar selectionMode='single'
                       onPageChange={onPageChange}
                       defaultValue='2024-03-11'/>);

      const button = screen.getByRole('button', {name: new RegExp(buttonName, 'i')})
      await user.click(button)

      expect(onPageChange).toHaveBeenCalledWith(expectedYear, expectedMonth)
    });
  });

  describe('<test prop:: renderDate>', () => {
    it('should render custom date', () => {
      const defaultRange: [Date, Date] = [
        new Date('2024-01-05'),
        new Date('2024-01-08'),
      ]
      const renderDate = jest.fn((date: Date) => <span>{date.getDate().toString().padStart(2, '0')}</span>)
      const { container } = render(<Calendar selectionMode='single'
        defaultValue={new Date(2024, 2, 11)}
 renderDate={renderDate}/>);

      expect(container).toMatchSnapshot()


    });
  });

  describe('<test prop:: allowClear>', () => {
    it('should allow clear by default', async () => {
      const user = userEvent.setup()
      render(<Calendar selectionMode='single'
                       defaultValue='2024-03-11'/>);

      const selectedCell = screen.getByRole('gridcell', {selected: true})

      expect(selectedCell).toHaveClass('mui-calendar-cell-selected')

      await user.click(selectedCell)

      expect(selectedCell).not.toHaveClass('mui-calendar-cell-selected')


    });

    it('should not allow clear when allowClear is false', async () => {
      const user = userEvent.setup()
      render(<Calendar selectionMode='single'
                       allowClear={false}
                       defaultValue='2024-03-11'/>);

      const selectedCell = screen.getByRole('gridcell', {selected: true})

      expect(selectedCell).toHaveClass('mui-calendar-cell-selected')

      await user.click(selectedCell)

      expect(selectedCell).toHaveClass('mui-calendar-cell-selected')


    });

  });

  describe('<test prop:: shouldDisableDate>', () => {
    it('should disabled date', () => {
      const shouldDisableDate = jest.fn(date => date.getDate() === 15);
      render(<Calendar selectionMode='single'
                       shouldDisableDate={shouldDisableDate}
                       defaultValue='2024-03-11'/>)

      expect(screen.getByRole('gridcell', {name: /15/i})).toHaveClass('mui-calendar-cell-disabled')
    });
  });

  describe('<test prop:: fieldid>', () => {
    it('should have fieldid for prev year button', () => {
      const {result: {current: fieldid}} = renderHook(useId)

      render(<Calendar fieldid={fieldid}/>)

      expect(screen.getByRole('button', {name: /prev year/i})).toHaveAttribute('fieldid', expect.stringContaining(fieldid))
    });

    it('should have fieldid for prev month button', () => {
      const {result: {current: fieldid}} = renderHook(useId)

      render(<Calendar fieldid={fieldid}/>)

      expect(screen.getByRole('button', {name: /prev month/i})).toHaveAttribute('fieldid', expect.stringContaining(fieldid))
    })

    it('should have fieldid for next month button ', () => {
      const {result: {current: fieldid}} = renderHook(useId)

      render(<Calendar fieldid={fieldid}/>)

      expect(screen.getByRole('button', {name: /next month/i})).toHaveAttribute('fieldid', expect.stringContaining(fieldid))
    });

    it('should have fieldid for next year button ', () => {
      const {result: {current: fieldid}} = renderHook(useId)

      render(<Calendar fieldid={fieldid}/>)

      expect(screen.getByRole('button', {name: /next year/i})).toHaveAttribute('fieldid', expect.stringContaining(fieldid))
    })
  });

  describe('<test prop:: clsPrefix>', () => {
    it('should have clsPrefix mui by default', () => {
      const {container} = render(<Calendar/>);

      expect(container.firstChild).toHaveClass(/^mui/i)
    });

    it('should have custom clasPrefix', () => {
      const {container} = render(<Calendar clsPrefix='test'/>);
      expect(container.firstChild).toHaveClass(/^test/i)
    });
  });


  it('should jump to a month, <test prop:: ref>', () => {
    const App = () => {
      const ref = useCallback((calendarRef: CalendarRef) => {
        calendarRef?.jumpTo({year: 2024, month: 4})
      }, [])
      return <Calendar ref={ref}/>
    }

    render(<App/>)

    const expectedResult = `2024年4月`
    expect(screen.getByText(expectedResult)).toBeInTheDocument()

  });


});
