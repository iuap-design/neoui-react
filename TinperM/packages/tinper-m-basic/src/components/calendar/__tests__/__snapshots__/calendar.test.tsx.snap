// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Component <test prop:: renderDate> should render custom date 1`] = `
<div>
  <div
    class="mui-calendar"
  >
    <div
      class="mui-calendar-header"
    >
      <a
        aria-label="Go to prev year"
        class="mui-calendar-arrow-button mui-calendar-arrow-button-left-year"
        fieldid="undefined-year-subtract-btn"
        role="button"
      >
        <svg
          height="1em"
          viewBox="0 0 44 44"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <g
              transform="translate(-24.000000, -22.000000)"
            >
              <g
                transform="translate(24.000000, 22.000000)"
              >
                <rect
                  height="44"
                  width="44"
                  x="0"
                  y="0"
                />
                <g
                  fill="currentColor"
                  fill-rule="nonzero"
                  transform="translate(7.000000, 4.000000)"
                >
                  <path
                    d="M19.4833058,2.71985611 L3.53051139,17.0699744 C3.0173831,17.5315665 2.97522952,18.3220903 3.43630803,18.8357433 L3.43630796,18.8357432 C3.46601289,18.8688164 3.49745845,18.9002801 3.53051133,18.9300007 L19.4833057,33.2801611 C20.1234001,33.8559077 20.1759552,34.8420707 19.6007967,35.4827774 C19.0256382,36.1235263 18.0404824,36.1761351 17.400388,35.6003885 L1.44759367,21.2502703 L1.4475933,21.25027 C1.33208743,21.1463692 1.22220259,21.036372 1.11840792,20.920748 C-0.49302969,19.1256817 -0.345639536,16.3628317 1.4475933,14.7497465 L17.4003877,0.399628282 C18.0404821,-0.176160428 19.0256378,-0.123509422 19.6007963,0.517239417 C20.1759548,1.1579461 20.1233997,2.14410915 19.4833053,2.7198557 L19.4833058,2.71985611 Z"
                  />
                  <path
                    d="M19.5305114,17.0699744 C19.0173831,17.5315665 18.9752295,18.3220903 19.436308,18.8357433 C19.4660129,18.8688164 19.4974585,18.9002801 19.5305113,18.9300007 L29.4833057,27.2801611 C30.1234001,27.8559077 30.1759552,28.8420707 29.6007967,29.4827774 C29.0256382,30.1235263 28.0404824,30.1761351 27.400388,29.6003885 L17.4475937,21.2502703 C17.3320874,21.1463692 17.2222026,21.036372 17.1184079,20.920748 C15.5069703,19.1256817 15.6543605,16.3628317 17.4475933,14.7497465 L27.4003877,6.39962828 C28.0404821,5.82383957 29.0256378,5.87649058 29.6007963,6.51723942 C30.1759548,7.1579461 30.1233997,8.14410915 29.4833053,8.7198557 L19.5305114,17.0699744 Z"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </a>
      <a
        aria-label="Go to prev month"
        class="mui-calendar-arrow-button mui-calendar-arrow-button-left-month"
        fieldid="undefined-month-subtract-btn"
        role="button"
      >
        <svg
          height="1em"
          viewBox="0 0 44 44"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <g
              transform="translate(-100.000000, -22.000000)"
            >
              <g
                transform="translate(100.000000, 22.000000)"
              >
                <rect
                  height="44"
                  width="44"
                  x="0"
                  y="0"
                />
                <g
                  fill="currentColor"
                  fill-rule="nonzero"
                  transform="translate(12.000000, 4.000000)"
                >
                  <path
                    d="M19.4833058,2.71985611 L3.53051139,17.0699744 C3.0173831,17.5315665 2.97522952,18.3220903 3.43630803,18.8357433 L3.43630796,18.8357432 C3.46601289,18.8688164 3.49745845,18.9002801 3.53051133,18.9300007 L19.4833057,33.2801611 C20.1234001,33.8559077 20.1759552,34.8420707 19.6007967,35.4827774 C19.0256382,36.1235263 18.0404824,36.1761351 17.400388,35.6003885 L1.44759367,21.2502703 L1.4475933,21.25027 C1.33208743,21.1463692 1.22220259,21.036372 1.11840792,20.920748 C-0.49302969,19.1256817 -0.345639536,16.3628317 1.4475933,14.7497465 L17.4003877,0.399628282 C18.0404821,-0.176160428 19.0256378,-0.123509422 19.6007963,0.517239417 C20.1759548,1.1579461 20.1233997,2.14410915 19.4833053,2.7198557 L19.4833058,2.71985611 Z"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </a>
      <div
        class="mui-calendar-title"
      >
        2024年3月
      </div>
      <a
        aria-label="Go to next month"
        class="mui-calendar-arrow-button mui-calendar-arrow-button-right mui-calendar-arrow-button-right-month"
        fieldid="undefined-month-add-btn"
        role="button"
      >
        <svg
          height="1em"
          viewBox="0 0 44 44"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <g
              transform="translate(-100.000000, -22.000000)"
            >
              <g
                transform="translate(100.000000, 22.000000)"
              >
                <rect
                  height="44"
                  width="44"
                  x="0"
                  y="0"
                />
                <g
                  fill="currentColor"
                  fill-rule="nonzero"
                  transform="translate(12.000000, 4.000000)"
                >
                  <path
                    d="M19.4833058,2.71985611 L3.53051139,17.0699744 C3.0173831,17.5315665 2.97522952,18.3220903 3.43630803,18.8357433 L3.43630796,18.8357432 C3.46601289,18.8688164 3.49745845,18.9002801 3.53051133,18.9300007 L19.4833057,33.2801611 C20.1234001,33.8559077 20.1759552,34.8420707 19.6007967,35.4827774 C19.0256382,36.1235263 18.0404824,36.1761351 17.400388,35.6003885 L1.44759367,21.2502703 L1.4475933,21.25027 C1.33208743,21.1463692 1.22220259,21.036372 1.11840792,20.920748 C-0.49302969,19.1256817 -0.345639536,16.3628317 1.4475933,14.7497465 L17.4003877,0.399628282 C18.0404821,-0.176160428 19.0256378,-0.123509422 19.6007963,0.517239417 C20.1759548,1.1579461 20.1233997,2.14410915 19.4833053,2.7198557 L19.4833058,2.71985611 Z"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </a>
      <a
        aria-label="Go to next year"
        class="mui-calendar-arrow-button mui-calendar-arrow-button-right mui-calendar-arrow-button-right-year"
        fieldid="undefined-year-add-btn"
        role="button"
      >
        <svg
          height="1em"
          viewBox="0 0 44 44"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <g
              transform="translate(-24.000000, -22.000000)"
            >
              <g
                transform="translate(24.000000, 22.000000)"
              >
                <rect
                  height="44"
                  width="44"
                  x="0"
                  y="0"
                />
                <g
                  fill="currentColor"
                  fill-rule="nonzero"
                  transform="translate(7.000000, 4.000000)"
                >
                  <path
                    d="M19.4833058,2.71985611 L3.53051139,17.0699744 C3.0173831,17.5315665 2.97522952,18.3220903 3.43630803,18.8357433 L3.43630796,18.8357432 C3.46601289,18.8688164 3.49745845,18.9002801 3.53051133,18.9300007 L19.4833057,33.2801611 C20.1234001,33.8559077 20.1759552,34.8420707 19.6007967,35.4827774 C19.0256382,36.1235263 18.0404824,36.1761351 17.400388,35.6003885 L1.44759367,21.2502703 L1.4475933,21.25027 C1.33208743,21.1463692 1.22220259,21.036372 1.11840792,20.920748 C-0.49302969,19.1256817 -0.345639536,16.3628317 1.4475933,14.7497465 L17.4003877,0.399628282 C18.0404821,-0.176160428 19.0256378,-0.123509422 19.6007963,0.517239417 C20.1759548,1.1579461 20.1233997,2.14410915 19.4833053,2.7198557 L19.4833058,2.71985611 Z"
                  />
                  <path
                    d="M19.5305114,17.0699744 C19.0173831,17.5315665 18.9752295,18.3220903 19.436308,18.8357433 C19.4660129,18.8688164 19.4974585,18.9002801 19.5305113,18.9300007 L29.4833057,27.2801611 C30.1234001,27.8559077 30.1759552,28.8420707 29.6007967,29.4827774 C29.0256382,30.1235263 28.0404824,30.1761351 27.400388,29.6003885 L17.4475937,21.2502703 C17.3320874,21.1463692 17.2222026,21.036372 17.1184079,20.920748 C15.5069703,19.1256817 15.6543605,16.3628317 17.4475933,14.7497465 L27.4003877,6.39962828 C28.0404821,5.82383957 29.0256378,5.87649058 29.6007963,6.51723942 C30.1759548,7.1579461 30.1233997,8.14410915 29.4833053,8.7198557 L19.5305114,17.0699744 Z"
                  />
                </g>
              </g>
            </g>
          </g>
        </svg>
      </a>
    </div>
    <div
      class="mui-calendar-mark"
    >
      <div
        class="mui-calendar-mark-cell"
      >
        日
      </div>
      <div
        class="mui-calendar-mark-cell"
      >
        一
      </div>
      <div
        class="mui-calendar-mark-cell"
      >
        二
      </div>
      <div
        class="mui-calendar-mark-cell"
      >
        三
      </div>
      <div
        class="mui-calendar-mark-cell"
      >
        四
      </div>
      <div
        class="mui-calendar-mark-cell"
      >
        五
      </div>
      <div
        class="mui-calendar-mark-cell"
      >
        六
      </div>
    </div>
    <div
      class="mui-calendar-cells"
    >
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            25
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            26
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            27
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            28
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            29
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            01
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            02
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            03
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            04
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            05
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            06
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            07
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            08
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            09
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            10
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        aria-selected="true"
        class="mui-calendar-cell mui-calendar-cell-selected mui-calendar-cell-selected-begin mui-calendar-cell-selected-end mui-calendar-cell-selected-single"
        fieldid="undefined-selected-end"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            11
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            12
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            13
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            14
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            15
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            16
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            17
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            18
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            19
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            20
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            21
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            22
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            23
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            24
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            25
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            26
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            27
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            28
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            29
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            30
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            31
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            01
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            02
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            03
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            04
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            05
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
      <div
        class="mui-calendar-cell mui-calendar-cell-disabled"
        role="gridcell"
      >
        <div
          class="mui-calendar-cell-top"
        >
          <span>
            06
          </span>
        </div>
        <div
          class="mui-calendar-cell-bottom"
        />
      </div>
    </div>
  </div>
</div>
`;
